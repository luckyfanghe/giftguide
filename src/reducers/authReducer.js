// @flow
import types from "../actions/types";

const initialState = {
  sidebarOpened: true,
  sidebarStyle: "login",
  authState: ""
};
function authReducer(state = initialState, action) {
  switch (action.type) {
    case types.AUTH.OPENSIDEBAR:
      return {
        ...state,
        sidebarOpened: true,
        sidebarStyle: action.sidebarStyle
      };
    case types.AUTH.CLOSESIDEBAR:
      return {
        ...state,
        sidebarOpened: false,
        sidebarStyle: action.sidebarStyle
      };
    case types.AUTH.DOINGAUTH:
      return {
        ...state,
        authState: action.state
      };
    default:
      return {
        ...state
      };
  }
}

export default authReducer;
