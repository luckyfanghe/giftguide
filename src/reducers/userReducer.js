// @flow
import types from "../actions/types";

const initialState = {
  user: null,
  userState: "",
  userStateDes: ""
};
function userReducer(state = initialState, action) {
  switch (action.type) {
    case types.USER.UPDATE:
      return {
        ...state,
        user: action.userInfo
      };
    case types.USER.LOGINFAILED:
      return {
        ...state,
        userState: "login failed",
        userStateDes: action.state
      };
    case types.USER.LOGINSUCCESS:
      return {
        ...state,
        userState: "login success",
        userStateDes: ""
      };
    case types.USER.SIGNUPFAILED:
      return {
        ...state,
        userState: "signup failed",
        userStateDes: action.state
      };
    case types.USER.SIGNUPSUCCESS:
      return {
        ...state,
        userState: "signup success",
        userStateDes: ""
      };
    default:
      return {
        ...state
      };
  }
}

export default userReducer;
