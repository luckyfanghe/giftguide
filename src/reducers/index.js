import { combineReducers } from "redux";
import { connectRouter } from "connected-react-router";

import userReducer from "./userReducer";
import authReducer from "./authReducer";

export default combineReducers({
  userReducer,
  authReducer
});
