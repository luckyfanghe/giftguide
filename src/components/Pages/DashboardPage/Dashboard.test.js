import React from "react";
import { render, fireEvent } from "@testing-library/react";
import renderer from "react-test-renderer";
import Dashboard from "./Dashboard";

import { Provider } from "react-redux";
import { composeWithDevTools } from "redux-devtools-extension";
import { createStore, applyMiddleware, compose } from "redux";
import createSagaMiddleware from "redux-saga";
import reducers from "../../../reducers";
import rootSaga from "../../../sagas";

const sagaMiddleware = createSagaMiddleware();
const enhancers = composeWithDevTools(applyMiddleware(sagaMiddleware));

const store = createStore(reducers, enhancers);

sagaMiddleware.run(rootSaga);

it("Dashboard renders corrently", () => {
  const { queryByTestId } = render(
    <Provider store={store}>
      <Dashboard />
    </Provider>
  );

  expect(queryByTestId("Dashboard")).toBeTruthy();
});

describe("Dashboard", () => {
  test("Dashboard snapshot renders ", () => {
    const component = renderer.create(
      <Provider store={store}>
        <Dashboard />
      </Provider>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
