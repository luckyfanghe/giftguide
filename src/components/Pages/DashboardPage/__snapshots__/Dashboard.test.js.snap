// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard Dashboard snapshot renders  1`] = `
<div
  className="sc-AykKK ePxKVu"
  data-testid="Dashboard"
>
  <div
    className="sc-fzXfLO bYoUJB"
  >
    <div>
      SIMPLIFY GIFTING AND ENSURE DELIGHTED
    </div>
    <div>
      FACES ON SPECIAL OCCASIONS
    </div>
  </div>
  <div
    className="sc-fzXfLP oogYd"
  >
    <div
      className="sc-AykKC eZouOZ"
      data-testid="Card"
      id="card-1"
    >
      <div
        className="sc-AykKD jerNDl"
      >
        CALENDAR
      </div>
      <div
        className="sc-AykKF hnEzuj"
      >
        <div
          className="sc-AykKG fVihnv"
          id="des-1"
        >
          gift guide creates a customized
        </div>
        <div
          className="sc-AykKG fVihnv"
          id="des-2"
        >
          calendar for you with all your loved
        </div>
        <div
          className="sc-AykKG fVihnv"
          id="des-3"
        >
          one's most important dates, in one
        </div>
        <div
          className="sc-AykKG fVihnv"
          id="des-4"
        >
          easy location
        </div>
      </div>
      <button
        className="sc-AykKE dosAzx"
        type="button"
      >
         
        ADD FRIENDS
         
      </button>
    </div>
    <div
      className="sc-AykKC eZouOZ"
      data-testid="Card"
      id="card-2"
    >
      <div
        className="sc-AykKD jerNDl"
      >
        WISH LIST
      </div>
      <div
        className="sc-AykKF hnEzuj"
      >
        <div
          className="sc-AykKG fVihnv"
          id="des-5"
        >
          gift guide users set up profiles
        </div>
        <div
          className="sc-AykKG fVihnv"
          id="des-6"
        >
          containing their favorite stores, wish
        </div>
        <div
          className="sc-AykKG fVihnv"
          id="des-7"
        >
          lists and items they love, so you no
        </div>
        <div
          className="sc-AykKG fVihnv"
          id="des-8"
        >
          longer have to ask for ideas or sizes
        </div>
      </div>
      <button
        className="sc-AykKE dosAzx"
        type="button"
      >
         
        ADD TO WISH LIST
         
      </button>
    </div>
    <div
      className="sc-AykKC eZouOZ"
      data-testid="Card"
      id="card-3"
    >
      <div
        className="sc-AykKD jerNDl"
      >
        REMINDERS
      </div>
      <div
        className="sc-AykKF hnEzuj"
      >
        <div
          className="sc-AykKG fVihnv"
          id="des-9"
        >
          gift guide sends you reminders before
        </div>
        <div
          className="sc-AykKG fVihnv"
          id="des-10"
        >
          your loved ones' events with their
        </div>
        <div
          className="sc-AykKG fVihnv"
          id="des-11"
        >
          wish lists and favorite stores so you
        </div>
        <div
          className="sc-AykKG fVihnv"
          id="des-12"
        >
          never miss an important date again
        </div>
      </div>
      <button
        className="sc-AykKE dosAzx"
        type="button"
      >
         
        SET UP REMINDERS
         
      </button>
    </div>
  </div>
  <div
    className="sc-fzXfLQ dhPgs"
  >
    <div
      className="sc-AykKH lrMCr"
      data-testid="Banner"
    >
      <img
        alt="banner"
        className="sc-AykKI kzpyli"
        src="banner2.webp"
      />
      <button
        className="sc-AykKJ crLLTO"
        type="button"
      >
        GET STARTED
      </button>
    </div>
    <div
      className="sc-AykKH lrMCr"
      data-testid="Banner"
    >
      <img
        alt="banner"
        className="sc-AykKI kzpyli"
        src="banner1.webp"
      />
      <button
        className="sc-AykKJ crLLTO"
        type="button"
      >
        BROWSE GIFT IDEAS
      </button>
    </div>
  </div>
</div>
`;
